test/puppetlabs/puppetdb/testutils/services.clj

```diff
@@ -352,3 +355,28 @@
   (until-consumed
    (fn []
      (pdb-client/submit-command-via-http! base-url cmd version payload))))
+
+(defn wait-for-server-processing
+  "Returns a truthy value indicating whether the wait was
+  successful (i.e. didn't time out).  Given a false or nil timeout,
+  waits forever."
+  ([server] (wait-for-server-processing server nil))
```

Comment:
OK, we were using it before, and I thought I'd preserve it in case someone did want to wait forever, but I suppose they can manage that themselves, or we  can add support later.